### ISSUE TITLE
Testing the Sdk

### ISSUE THREAD

**Issue by JohnGeorgiou** [helpcrunch/android-sdk-demo#27]
Hi, I'm testing the Android Sdk and I have some questions

**1. Driving a chat to  a specific Department** 
I have already created 3 departmemts A, B, & C, and I assigned a unique agent in each one.
Is it possible to drive a chat to a specific Department by code? So only an agent that belongs to this department to be able to respond.
At the moment the sdk provides the user a bunch of agents of any department.

**2. Avatar**
In my app I have registered users.
I successfuly create an HCUser, and passing textual info as Name, Phone etc
Is it possible to pass also the img url of the user to embed it in the chat?

**3. Attach file**
Is it possible to remove / hide the attach icon?
I will not permit the users to upload any files...

Thank you in advance

**Comment by TalbotGooday** [comment]
Hello! 
***1*** and ***2*** unfortunately, this is not possible now. But I will pass this idea on to our managers. Sounds interesting :100: 
***3.*** Yes, you can do this in the Android application settings: ***Settings - Android APPs - `Your android App` - Customization - 
File Attachments (uncheck if you want to disable this option)***
***4.*** It's a good question really. But I need to know exactly how your localization works. If you restart the application after changing the language, then in theory, everything should work fine if you've added the needed localized strings to your app for our SDK. Tell me please how you have implemented this functionality. Maybe I can come up with a solution

Thanks for your questions :)

**Comment by JohnGeorgiou** [comment]
Thank you for the fast reply,

**1** is the most important in my case.
I'll have **up to 10** individual departments, and I'll assign **one** agent to each of them. 
The agents in these departments have a very specific know-how and they can't exchange their positions. 
With this senario I have to set-up 10 different helpcrunch accounts, with different dashboards, but only one mobile app, losing any overall anaysis of the events, tracking events etc. 
And of course billing all these accounts its not afortable.
At least may be you can find a solution passing custom data (key value) to drive the chat **directly to the agent** and not to the department? Lets think about it.

The other issues ar minors.
 **4**, for reference I use this lib https://github.com/zeugma-solutions/locale-helper-android but I think that if you aim to do something about it you have to develop something more generic i.e. loading a specific string resources xml by code...

thank you again!

**Comment by JohnGeorgiou** [comment]
Hi again,
Visiting the API docs, I see that using the API you have the ability to assign a chat to a specific team member by its ID
https://docs.helpcrunch.com/rest-api-v1/update-chat-assignee-v1
I should appreciate if you embed this functionality in the sdk...

**Comment by TalbotGooday** [comment]
Hello again! 
I talked to the managers, tested the work with the library and here is the result:
1. > It’s not possible now. However, we have the access settings for each role is on our roadmap. Also, you can check chat distribution settings in a department. Go to web application: Settings -> Team -> Departments and move to department you want to assign automated chat assigning. The settings are in the bottom of a department’s page.
2. > It’s something that wasn’t requested before so there is no way you can do it now. But we can think about adding this to Rest API and mobile SDKs.

In addition, you can use our API to implement the functionality you need. Like getting agents list or assigning the agent to the chat :+1: 
I'll also check if we have the event to track the creation of a new chat. If we don't so we can add it for you

Also, about the `locale-helper-android`. 
I've tested the implementation with this lib and seems like it works well. I want to note that by default we do not have translation into languages other than English so you need to add them manually. Maybe this is the main reason why locale in the SDK did not changed. Because there are no any other localization except the English localization :thinking:

**Comment by JohnGeorgiou** [comment]
Thanks,
So, I'll abandon the SDK and I'll make some tests using the Api via Postman , to check if I can integrate the requested functionality.

**Comment by TalbotGooday** [comment]
Awesome! Write in this issue if you have additional questions :+1:

**Comment by JohnGeorgiou** [comment]
Hi again,
is the API available in trial version?
I've included the http headers with my Api Key , but I'got errors
  "code": 401,
  "message": "Invalid private key"

**Comment by TalbotGooday** [comment]
Yeah, I think is available. Could you please send me a cURL or someting like this. I need to know what exactly you are sending

**Comment by JohnGeorgiou** [comment]
I sent an email to you gmail account, thanks

**Comment by TalbotGooday** [comment]
I don't have an access to this account :sweat_smile: I will try to get to this letter, but if you want to speed up the process, then just send a request here. Without tokens, of course :)

**Comment by TalbotGooday** [comment]
I just now realized that you sent the request information to my personal account a week ago. Sorry 😅  Please, write if your question is relevant and I will see how I can help)
